cloudevents/emitter.go

```diff
@@ -0,0 +1,60 @@
+package cloudevents
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cloudevents/sdk-go/v2/client"
+	"github.com/cloudevents/sdk-go/v2/event"
+	"github.com/cloudevents/sdk-go/v2/protocol/http"
+	"github.com/cloudevents/sdk-go/v2/types"
+	"github.com/google/uuid"
+)
+
+type Emitter struct {
+	Endpoint    string
+	Source      string
+	Type        string
+	Id          string
+	Data        string
+	ContentType string
+}
+
+func NewEmitter() *Emitter {
+	return &Emitter{
+		Source:      "/boson/fn",
+		Type:        "boson.fn",
+		Id:          uuid.NewString(),
+		Data:        "",
+		ContentType: event.TextPlain,
+	}
+}
```

Comment:
One thing I have seen done, and found to be pleasant for these default values is to turn them into exported constants.  That has a few benefits, among them being the value is included in the auto-generated godoc package documentaiton.  It might be worth adopting if you like that as well:

```suggestion
const (
  // DefaultSource for emitted events.
  DefaultSource = "/boson/fn"
  
  // DefaultType of emitted events.
  DefaultType = "boson.fn"
)

type Emitter struct {
	Endpoint    string
	Source      string
	Type        string
	Id          string
	Data        string
	ContentType string
}

func NewEmitter() *Emitter {
	return &Emitter{
		Source:      DefaultSource,
		Type:        DefaultType,
		Id:          uuid.NewString(),
		Data:        "",
		ContentType: event.TextPlain,
	}
}
```